### PR TITLE
port naive_path_prediction from ROS1 to ROS2

### PR DESCRIPTION
**Pull request by nik-tier4**
edited CMakeLists.txt, package.xml, launch file
node.cpp & node.hpp files.

### CODE REVIEW

File: perception/object_recognition/prediction/naive_path_prediction/package.xml [hunk 1]
```diff
@@ -41,24 +41,14 @@
...
-  <build_export_depend>tf2</build_export_depend>
-  <exec_depend>autoware_perception_msgs</exec_depend>
-  <exec_depend>geometry_msgs</exec_depend>
-  <exec_depend>roscpp</exec_depend>
-  <exec_depend>tf2</exec_depend>
-
-
-  <!-- The export tag contains other, unspecified, tags -->
+  <buildtool_depend>ament_cmake</buildtool_depend>
+  <buildtool_depend>ament_cmake_auto</buildtool_depend>
```
Comment (sgermanserrano) [inline perception/object_recognition/prediction/naive_path_prediction/package.xml:45]: There is no need to have both here, having just `ament_cmake` will work fine

File: perception/object_recognition/prediction/naive_path_prediction/src/node.cpp [hunk 2]
```diff
@@ -22,29 +22,32 @@
 #include <tf2/convert.h>
 #include <tf2/transform_datatypes.h>
 #include <tf2_geometry_msgs/tf2_geometry_msgs.h>
-#include "geometry_msgs/PoseWithCovarianceStamped.h"
+#include "geometry_msgs/msg/pose_with_covariance_stamped.hpp"
 
-NaivePathPredictionNode::NaivePathPredictionNode() : nh_(""), pnh_("~")
+NaivePathPredictionNode::NaivePathPredictionNode() : Node("node")
```
Comment (sgermanserrano) [inline perception/object_recognition/prediction/naive_path_prediction/src/node.cpp:27]: Please use a more meaningful node name such as:
```suggestion
NaivePathPredictionNode::NaivePathPredictionNode() : Node("NaivePathPredictionNode")
```

**Review by sgermanserrano — CHANGES_REQUESTED** [review]

File: perception/object_recognition/prediction/naive_path_prediction/package.xml (hunk 1 above)
Comment (nik-tier4) [reply perception/object_recognition/prediction/naive_path_prediction/package.xml:44]: requested changes made

File: perception/object_recognition/prediction/naive_path_prediction/src/node.cpp (hunk 2 above)
Comment (nik-tier4) [reply perception/object_recognition/prediction/naive_path_prediction/src/node.cpp]: requested changes made.

**Review by jilaada — COMMENTED** [review]
LGTM, though if possible could you please fix the conflicts and maybe add the linter tests.

**Comment by TakaHoribe** [comment]
Merge this after CI is passed

**Pull request merged**